workspace/notebook/appeal_document.json

```diff
@@ -315,14 +276,64 @@
 					}
 				},
 				"source": [
-					"##### Produce Json formatted output"
+					"#####  Logging Execution Metadata to Azure Application Insights"
 				]
 			},
 			{
 				"cell_type": "code",
 				"source": [
-					"# Exit with the JSON result\n",
-					"mssparkutils.notebook.exit(app_insight_logger.generate_processing_results())"
+					"try:\n",
+					"    # Calculate execution duration\n",
+					"    duration_seconds = (end_exec_time - start_exec_time).total_seconds()\n",
+					"\n",
+					"    # Define activity type\n",
+					"    activity_type = f\"{mssparkutils.runtime.context['currentNotebookName']} Notebook\"\n",
+					"\n",
+					"    # Determine status and message\n",
+					"    stage = \"Success\" if not error_message else \"Failed\"\n",
+					"    status_message = (\n",
+					"        f\"Successfully loaded data into {table_name} table\"\n",
+					"        if not error_message\n",
+					"        else f\"Failed to load data from {table_name} table\"\n",
+					"    )\n",
+					"    status_code = \"200\" if stage == \"Success\" else \"500\"\n",
+					"\n",
+					"    # Prepare telemetry parameters\n",
+					"    params = {\n",
+					"        \"Stage\": stage,\n",
+					"        \"PipelineName\": PipelineName,\n",
+					"        \"PipelineRunID\": PipelineRunID,\n",
+					"        \"StartTime\": start_exec_time.isoformat(),\n",
+					"        \"EndTime\": end_exec_time.isoformat(),\n",
+					"        \"Inserts\": insert_count,\n",
+					"        \"Updates\": update_count,\n",
+					"        \"Deletes\": delete_count,\n",
+					"        \"ErrorMessage\": error_message,\n",
+					"        \"StatusMessage\": status_message,\n",
+					"        \"PipelineTriggerID\": PipelineTriggerID,\n",
+					"        \"PipelineTriggerName\": PipelineTriggerName,\n",
+					"        \"PipelineTriggerType\": PipelineTriggerType,\n",
+					"        \"PipelineTriggeredbyPipelineName\": PipelineTriggeredbyPipelineName,\n",
+					"        \"PipelineTriggeredbyPipelineRunID\": PipelineTriggeredbyPipelineRunID,\n",
+					"        \"PipelineExecutionTimeInSec\": duration_seconds,\n",
+					"        \"ActivityType\": activity_type,\n",
+					"        \"DurationSeconds\": duration_seconds,\n",
+					"        \"StatusCode\": status_code,\n",
+					"        \"AppInsCustomEventName\": \"ODW_Master_Pipeline_Logs\"\n",
+					"    }\n",
+					"\n",
+					"    # Send telemetry asynchronously\n",
+					"    send_telemetry_to_app_insights(params)\n",
+					"\n",
+					"    # Raise error if execution failed\n",
+					"    if error_message:\n",
+					"        print(f\"Notebook Failed for load {table_name} : {error_message}\")\n",
+					"        raise RuntimeError(f\"Notebook Failed due to error in {table_name} Table: {error_message}\")\n",
+					"\n",
+					"except RuntimeError as e:\n",
+					"    print(str(e))\n",
+					"    import sys\n",
+					"    sys.exit(1)"
```

Comment:
Based on our conversation yesterday, could you not just do
```
					"    print(str(e))\n",
					"    import sys\n",
					"    sys.exit(1)"
```
directly, instead of raising a runtime error with a try/catch?